File: backend/core/responses/baseVideoResponse.py
from typing import Literal, Optional, List
from pydantic import BaseModel

from backend.core.responses.baseArtistResponse import BaseArtistResponse


class BaseVideoResponse(BaseModel):
    """Base response model for video information."""

    type: Literal["video"] = "video"
    provider: str
    publicId: str
    url: str
    name: str
    videoSrc: Optional[str] = None
    audioSrc: Optional[str] = None
    imageUrl: str
    duration_ms: int | None
    artists: List[BaseArtistResponse]
    downloaded: bool

File: frontend/packages/dto/baseVideoResponse.ts
import { BaseArtistResponseSchema } from "@/dto";
import { z } from "zod";

export const BaseVideoResponseSchema = z.object({
    type: z.union([z.literal("video")]).default("video"),
    provider: z.string(),
    publicId: z.string(),
    url: z.string(),
    name: z.string(),
    videoSrc: z.string().nullable(),
    audioSrc: z.string().nullable(),
    imageUrl: z.string(),
    duration_ms: z.number().nullable(),
    artists: z.array(z.lazy(() => BaseArtistResponseSchema)),
    downloaded: z.boolean(),
});

export type BaseVideoResponse = z.infer<typeof BaseVideoResponseSchema>;
